gateway/ln-gateway/src/lib.rs

```diff
@@ -295,12 +295,16 @@ impl LnGateway {
             .get_deposit_address()
     }
 
-    async fn handle_deposit_msg(&self, deposit: DepositPayload) -> Result<TransactionId> {
-        let rng = rand::rngs::OsRng;
-        self.federation_client
-            .peg_in(deposit.txout_proof, deposit.transaction, rng)
+    async fn handle_deposit_msg(&self, payload: DepositPayload) -> Result<TransactionId> {
+        let DepositPayload {
+            txout_proof,
+            transaction,
+            federation_id,
```

Comment:
Having seen the same pattern a few times now, could we extract the `federation_id` into another struct that contains the ID and the actual request?